src/main/java/org/dita/dost/module/MergeDitavalModule.java

```diff
@@ -0,0 +1,277 @@
+                } else {
+                    usingDitavalInput = basedir.toURI().resolve(ditavalInput);
+                }
+                if (toFile(usingDitavalInput).exists()) {
+                    ditavalFiles.add(toFile(usingDitavalInput));
+                } else {
+                    logger.error(
+                            MessageUtils.getInstance().getMessage("DOTJ071E", usingDitavalInput.toString()).toString());
+                }
+            }
+        }
+
+        // Set the mapDir
+        job.setInputFile(rootFile);
+    }
+
+    /**
+     * Update base directory and prefix based on uplevels.
+     */
+    private void updateBaseDirectory() {
+        for (int i = uplevels; i > 0; i--) {
+            baseInputDir = baseInputDir.resolve("..");
+        }
+    }
+    
+    /**
+     * Set external property with the name of the merged file.
+     */
+    private void setMergedProperty() {
+        job.setProperty("dita.input.valfile", new File(job.tempDir, FILE_NAME_MERGED_DITAVAL).toString());
+    }
+
+    private void writeMergedDitaval() throws DITAOTException {
+        XMLStreamWriter export = null;
+        try (OutputStream exportStream = new FileOutputStream(new File(job.tempDir, FILE_NAME_MERGED_DITAVAL))) {
+            export = XMLOutputFactory.newInstance().createXMLStreamWriter(exportStream, "UTF-8");
+            export.writeStartDocument();
+            export.writeStartElement("val");
+            export.writeNamespace(DITA_OT_NS_PREFIX, DITA_OT_NS);
+            for (File curDitaVal : ditavalFiles) {
+                final DocumentBuilder ditavalbuilder = XMLUtils.getDocumentBuilder();
+                final Document doc = ditavalbuilder.parse(curDitaVal);
+                final Element ditavalRoot = doc.getDocumentElement();
+                final NodeList rootChildren = ditavalRoot.getChildNodes();
+                logger.debug("Writing conditions from ditaval: " + curDitaVal);
+                writeConditions(export, rootChildren, curDitaVal.toURI().resolve("."));
+            }
+            export.writeEndElement();
+            export.writeEndDocument();
+        } catch (final IOException e) {
+            throw new DITAOTException("Failed to merge ditaval files: " + e.getMessage(), e);
+        } catch (final XMLStreamException e) {
+            throw new DITAOTException("Failed to serialize merged ditaval file: " + e.getMessage(), e);
+        } catch (final SAXException e) {
+            throw new DITAOTException("Failed to parse ditaval file: " + e.getMessage(), e);
+        } finally {
+            if (export != null) {
+                try {
+                    export.close();
+                } catch (final XMLStreamException e) {
+                    logger.error("Failed to close merged ditaval file: " + e.getMessage(), e);
+                }
+            }
+        }
+
+    }
+
+    private void writeConditions(XMLStreamWriter export, NodeList conditionElements, URI ditavalDirectory) {
```

Comment:
TODO: This should be generalized to be a DOM to XML stream utility function and moved to XMLUtils.